Assets/Scripts/RoadBuilder.cs

```diff
@@ -0,0 +1,179 @@
+﻿using System;
+using System.Collections.Generic;
+using UnityEngine;
+using Zenject;
+
+namespace Road
+{
+
+	public enum RoadType : int
+	{
+		DeadEnd = 1,
+		Straight = 2,
+		Turn = 3,
+		TJunction = 4,
+		CrossRoads = 5
+	}
+
+	public class RoadSegment
+	{
+		public RoadType roadType;
+		public Coordinate coords;
+		public Direction direction;
+
+		public RoadSegment (RoadType roadType, Coordinate coords, Direction direction)
+		{
+			this.roadType = roadType;
+			this.coords = coords;
+			this.direction = direction;
+		}
+	}
+
+	public class RoadBuilder : MonoBehaviour
+	{
+		public RoadSegment[] CreateRoadSegments (PathNode[] nodes)
+		{
+			RoadSegment[] roadSegments = new RoadSegment[nodes.Length];
+			for (int currentIndex = 0; currentIndex < nodes.Length; currentIndex++) {
+				roadSegments [currentIndex] = CreateRoadSegment (nodes, currentIndex);
+			}
+			return roadSegments;
+		}
+
+		private RoadSegment CreateRoadSegment (PathNode[] nodes, int currentNodeIndex)
+		{
+			PathNode node = nodes [currentNodeIndex];
+			switch (node.connectedNodes.Length) {
+			case 1:
```

Comment:
@mrniket Commented in the last review about using the enum names here rather than ints.

I don't think I can do that yet as we haven't mapped to the different RoadSegments at this point, we're just looking at the number of connected nodes here.

Reply:
I see, lets leave it for something to solve in the future